stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml

```diff
@@ -89,10 +92,53 @@
         >
             <attribute name="t-if">move_line.id</attribute>
         </xpath>
-        <!-- Remove "All items could not be shipped..." sentence -->
+        <!-- Remove "All items could not be shipped..." sentence
+            Substitute it by a message indicating if all goods have been delivered. -->
         <xpath expr='//div[hasclass("page")]/p[last()]' position="attributes">
             <attribute name="t-if">False</attribute>
         </xpath>
+        <xpath expr='//div[hasclass("page")]/p[last()]' position="after">
+            <t t-if="o.picking_type_id.code == 'outgoing'">
+                <p>Remaining to deliver:</p>
+                <t t-if="o.demand_is_completed()">
```

Comment:
Implementations of `demand_is_completed` and `get_remaining_to_deliver_data` are not consistent.
I would drop the first one and use `t-if="not pending_moves"`